src/hooks/screen/useChatScreenHook.ts

```diff
@@ -124,17 +127,62 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       }
 
       let response;
+
+      const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
+        if (item.type === 'image') {
+          const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+          return {
+            ...item,
+            asset_url: uploadedImageUrl.data.url,
+            thumb_url: uploadedImageUrl.data.url
+          };
+        }
+        if (item.type === 'video') {
+          const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+          const uploadedUrl = await ImageUtils.uploadFile(
+            item.video_path,
+            item.video_name,
+            item.video_type
+          );
+          return {
+            ...item,
+            asset_url: uploadedImageUrl.data.url,
+            thumb_url: uploadedImageUrl.data.url,
+            video_path: uploadedUrl.data.url
+          };
+        }
+        if (item.type === 'file') {
+          const uploadedUrl = await ImageUtils.uploadFile(
+            item.file_path,
+            item.file_name,
+            item.file_type
+          );
+          return {
+            ...item,
+            asset_url: uploadedUrl.data.url,
+            thumb_url: uploadedUrl.data.url,
+            file_path: uploadedUrl.data.url
+          };
+        }
+
+        return item;
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code handles the uploading of different types of attachments (image, video, file) and returns an array of promises. Each promise resolves to an object that includes the uploaded URL of the attachment. This is a good approach to handle multiple attachments concurrently. However, there's no error handling in case any of the uploads fail. Consider adding a try-catch block inside the map function to handle any errors during the upload process.
```diff
+       const allAttachmentPromises: Promise<void>[] = attachments.map(async (item: any) => {
+         try {
+           if (item.type === 'image') {
+             const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+             return {
+               ...item,
+               asset_url: uploadedImageUrl.data.url,
+               thumb_url: uploadedImageUrl.data.url
+             };
+           }
+           if (item.type === 'video') {
+             const uploadedImageUrl = await ImageUtils.uploadImage(item.asset_url);
+             const uploadedUrl = await ImageUtils.uploadFile(
+               item.video_path,
+               item.video_name,
+               item.video_type
+             );
+             return {
+               ...item,
+               asset_url: uploadedImageUrl.data.url,
+               thumb_url: uploadedImageUrl.data.url,
+               video_path: uploadedUrl.data.url
+             };
+           }
+           if (item.type === 'file') {
+             const uploadedUrl = await ImageUtils.uploadFile(
+               item.file_path,
+               item.file_name,
+               item.file_type
+             );
+             return {
+               ...item,
+               asset_url: uploadedUrl.data.url,
+               thumb_url: uploadedUrl.data.url,
+               file_path: uploadedUrl.data.url
+             };
+           }
+           return item;
+         } catch (error) {
+           console.error(`Error uploading ${item.type}: `, error);
+           throw error;
+         }
+       });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->